File: jaxmpp-jvm/src/main/java/org/tigase/jaxmpp/core/xml/parser/SimpleHandler.java
/*
 * SimpleHandler.java
 *
 * Tigase Jabber/XMPP XML Tools
 * Copyright (C) 2004-2017 "Tigase, Inc." <office@tigase.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, version 3 of the License.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. Look for COPYING file in the top folder.
 * If not, see http://www.gnu.org/licenses/.
 */

package org.tigase.jaxmpp.core.xml.parser;

/**
 * <code>SimpleHandler</code> - parser handler interface for event driven parser. It is very simplified version of
 * <code>org.xml.sax.ContentHandler</code> interface created for <code>SimpleParser</code> needs. It allows to receive
 * events like start element (with element attributes), end element, element cdata, other XML content and error event if
 * XML error found.
 * <p> Created: Sat Oct  2 00:00:08 2004 </p>
 *
 * @author <a href="mailto:artur.hefczyc@tigase.org">Artur Hefczyc</a>
 * @version $Rev$
 * @see SimpleParser
 */

public interface SimpleHandler {

	void elementCData(StringBuilder cdata);

	boolean endElement(StringBuilder name);

	void error(String errorMessage);

	void otherXML(StringBuilder other);

	Object restoreParserState();

	void saveParserState(Object state);

	void startElement(StringBuilder name, StringBuilder[] attr_names, StringBuilder[] attr_values);

}// SimpleHandler

File: src/commonMain/kotlin/tigase/halcyon/core/xml/parser/SimpleHandler.kt
/*
 * Tigase Halcyon XMPP Library
 * Copyright (C) 2018 Tigase, Inc. (office@tigase.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, version 3 of the License.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. Look for COPYING file in the top folder.
 * If not, see http://www.gnu.org/licenses/.
 */
package tigase.halcyon.core.xml.parser

interface SimpleHandler {

	fun error(errorMessage: String)

	fun startElement(name: String, attr_names: Array<String?>?, attr_values: Array<String?>?)

	fun elementCData(cdata: String)

	fun endElement(name: String): Boolean

	fun otherXML(other: String)

	fun saveParserState(state: Any?)

	fun restoreParserState(): Any?

}